src/mbgl/renderer/layers/render_fill_layer.cpp

```diff
@@ -452,21 +385,56 @@ void RenderFillLayer::update(gfx::ShaderRegistry& shaders,
     for (const RenderTile& tile : *renderTiles) {
         const auto& tileID = tile.getOverscaledTileID();
 
-        // If we already have drawables for this tile, skip.
-        if (tileLayerGroup->getDrawableCount(renderPass, tileID) > 0) {
-            continue;
-        }
-
         const LayerRenderData* renderData = getRenderDataForPass(tile, renderPass);
         if (!renderData) {
             removeTile(renderPass, tileID);
             continue;
         }
 
         auto& bucket = static_cast<FillBucket&>(*renderData->bucket);
+        const auto& paintPropertyBinders = bucket.paintPropertyBinders.at(getID());
+
         const auto& evaluated = getEvaluated<FillLayerProperties>(renderData->layerProperties);
         const auto& crossfade = getCrossfade<FillLayerProperties>(renderData->layerProperties);
 
+        const auto& fillPatternValue = evaluated.get<FillPattern>().constantOr(Faded<expression::Image>{"", ""});
+        const auto patternPosA = tile.getPattern(fillPatternValue.from.id());
+        const auto patternPosB = tile.getPattern(fillPatternValue.to.id());
+        paintPropertyBinders.setPatternParameters(patternPosA, patternPosB, crossfade);
+
+        const FillInterpolateUBO interpolateUBO = {
+            /* .color_t = */ std::get<0>(paintPropertyBinders.get<FillColor>()->interpolationFactor(state.getZoom())),
```

Comment:
state.getZoom() returns a double, we need to static_cast it to float to suppress implicit conversion warnings.